packages/angular_devkit/core/src/json/schema/interface.ts

```diff
@@ -74,7 +74,7 @@ export interface SchemaFormat {
 }
 
 export interface SmartDefaultProvider<T> {
-  (schema: JsonObject): T | Observable<T>;
+  (schema: JsonObject): T | Observable<T> | Promise<T>;
```

Comment:
This makes smart default providers async. I'd prefer we stick to observables.

Reply:
The return value is coming from third party code.  The observable can't be assumed to be synchronously completed.  Not having the promise as an option forces consumers to directly depend on rxjs and wrap all promise generating code in a `from`.  It also prevents the direct use of an async function which is the use case here.

The schema validation itself also needs to be assumed async since ajv can potentially complete either synchronously or asynchronously.